worker/rhpv3.go

```diff
@@ -211,19 +212,21 @@ func (t *transportV3) DialStream(ctx context.Context) (*streamV3, error) {
 
 // transportPoolV3 is a pool of rhpv3.Transports which allows for reusing them.
 type transportPoolV3 struct {
-	mu   sync.Mutex
-	pool map[string]*transportV3
+	mu     sync.Mutex
+	dialer *fallbackDialer
```

Comment:
The dialer is not protected by the mutex, so we should move it above it to make that clear.